geps/gep-713.md

```diff
@@ -1244,91 +1244,35 @@ level. The implementations that support this policy attachment model will have
 the same behavior and semantics, although they may not be able to support
 attachment of all types of policy at all potential attachment points.
 
-### Apply Policies to Sections of a Resource (Future Extension)
-Although initially out of scope, it would be helpful to be able to target
-specific matches within nested objects. For example, it may be useful to attach
-policies to a specific Gateway listener or Route rule. This section explores
-what that could look like.
-
-Each Route rule or Gateway listener should be expanded with an optional name
-field. The target ref would be expanded with an optional sectionName field that
-could be used to refer to that specific section of the resource. It would refer
-to the following concepts on these resources:
-
-* Gateway.Listeners.Name
-* xRoute.Rules.Name
-* Service.Ports.Name
+### Apply Policies to Sections of a Resource
+Policies can target specific matches within nested objects. For instance, rather than
+applying a policy to the entire Gateway, we may want to attach it to a particular Gateway listener.
+
+To achieve this, an optional `sectionName` field can be set in the `targetRef` of a policy
+to refer to a specific listener within the target Gateway.
 
 ```yaml
-apiVersion: gateway.networking.k8s.io/v1alpha2
-kind: HTTPRoute
-metadata:
-  name: http-app-1
-  labels:
-    app: foo
-spec:
-  hostnames:
-  - "foo.com"
-  rules:
-  - name: bar
-    matches:
-    - path:
-        type: Prefix
-        value: /bar
-    forwardTo:
-    - serviceName: my-service1
-      port: 8080
----
 apiVersion: networking.acme.io/v1alpha2
-kind: RetryPolicy
+kind: AuthenticationPolicy
 metadata:
   name: foo
 spec:
-  maxRetries: 5
+  provider:
+    issuer: "https://accounts.google.com"
   targetRef:
     name: foo
     group: gateway.networking.k8s.io
-    kind: HTTPRoute
+    kind: Gateway
```

Comment:
Just curious why the example was changed from an HTTPRoute to Gateway (and from RetryPolicy to AuthenticationPolicy). Whatever is the reason, could we have both examples? Specially because you also have an example of adding a name to RouteRule (on line 1277)

Reply:
@sanjaypujare The first example is changed to Gateway because the current use case I want to address is the Gateway Listener, and we already has the name field in the Listener object.  The second example is used to demonstrate what we need to change(adding the name field) to apply policy to RouteRule as well.

Reply:
Thanks. With this change you included the RetryPolicy but not the HTTPRoute in the original doc which has the named section. From https://gateway-api.sigs.k8s.io/geps/gep-713/#apply-policies-to-sections-of-a-resource-future-extension I have the snippet as follows:

```
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http-app-1
  labels:
    app: foo
spec:
  hostnames:
  - "foo.com"
  rules:
  - name: bar
    matches:
    - path:
        type: Prefix
        value: /bar
    forwardTo:
    - serviceName: my-service1
      port: 8080
---
```

This is just above the RetryPolicy so it will be good to include (or rather restore) the above snippet before line 1277.